handlers/legacy.go

```diff
@@ -120,11 +120,17 @@ func (lh *Legacy) buildNotificationMessage(
 ) (*messaging.NotificationMessage, error) {
 	wrapMsg := "unable to build notification message"
 
+	// Determine the primary text of the message portion of the notification.
+	messageText := payload.Message
+	if messageText == "" {
+		messageText = payload.Subject
+	}
+
 	// The message portion of the request sent to the UI is a JSON object.
 	outgoingMessage := map[string]interface{}{
 		"id":        request.ID,
 		"timestamp": common.FormatTimestamp(request.TimeCreated),
-		"text":      payload.Message,
```

Comment:
Not relevant since we're changing this now, but I didn't notice there was a secondary `message` key in the response as `payload.message`.  That value is, obviously, also empty in the responses. As far as I can tell, we're not using `payload.message` and only checking either the `subject` or the `message.text` value.